docs/reference/model.rst

```diff
@@ -1,53 +1,123 @@
-.. automodule:: stravalib.model
-    :inherited-members: BaseModel
+============
+Model
+============
 
-.. If you use both automodule and autosummary you duplicate autodocs efforts
-.. Summary Functions
-.. -------------------------------
-.. .. autosummary::
-..    :toctree: api/
+.. currentmodule:: stravalib.model
+
+Summary Athlete Classes
+------------------------
+.. autosummary::
+   :toctree: api/
+   :recursive:
+
+    AthleteStats
+    MetaAthlete
+    SummaryAthlete
+    DetailedAthlete
+    AthletePrEffort
+    AthleteSegmentStats
+
+Summary Activity Classes
+-------------------------
+.. autosummary::
+    :toctree: api/
+    :recursive:
+
+    MetaActivity
+    SummaryActivity
+    DetailedActivity
+    ClubActivity
+    ActivityTotals
+    ActivityComment
+    Lap
+    Split
+    RelaxedActivityType
+    RelaxedSportType
+    ActivityZone
+    TimedZoneDistribution
+
+Club Classes
+-------------
+.. autosummary::
+   :toctree: api/
+
+   MetaClub
+   SummaryClub
+   DetailedClub
+
+Summary Functions
+------------------
+
+.. autosummary::
+    :toctree: api/
+
+    check_valid_location
+    lazy_property
+    naive_datetime
+
+
+Helper Classes
+----------------------
+.. autosummary::
+    :toctree: api/
+
+    BoundClientEntity
+
+Unit & Conversion Classes
+--------------------------
+.. autosummary::
+    :toctree: api/
 
-..    lazy_property
-..    check_valid_location
-..    naive_datetime
+    Distance
+    Duration
+    Timezone
+    Velocity
 
-.. Summary Classes
-.. -------------------------------
+Activity Photo Classes
+-------------------------
+Note: the activity photo classes are defined differently in the Strava spec.
+This section will likely need to be updated. These endpoints are not well
+documented.
+
+.. autosummary::
+    :toctree: api/
+
+    ActivityPhotoPrimary
+    ActivityPhotoMeta
+    ActivityPhoto
+
+Subscription Classes
+-------------------------
+.. autosummary::
+   :toctree: api/
+
+   Subscription
+   SubscriptionCallback
+   SubscriptionUpdate
+
+Effort / Performance Classes
+-----------------------------
+.. autosummary::
+   :toctree: api/
+
+   Split
+   BaseEffort
+   BestEffort
+
+.. Route / Segment Classes
+.. -------------------------
 .. .. autosummary::
 ..    :toctree: api/
 
-..    DeprecatedSerializableMixin
-..    BackwardCompatibilityMixin
-..    BoundClientEntity
 ..    LatLon
-..    Club
-..    Gear
-..    Bike
-..    Shoe
-..    ActivityTotals
-..    AthleteStats
-..    Athlete
-..    ActivityComment
-..    ActivityPhotoPrimary
-..    ActivityPhotoMeta
-..    ActivityPhoto
-..    ActivityKudos
-..    ActivityLap
-..    Map
-..    Split
-..    SegmentExplorerResult
-..    AthleteSegmentStats
-..    AthletePrEffort
+..    SummarySegment
 ..    Segment
```

Comment:
Commented out because i am adding docstrings to each one. Then i'll double check the entire list. The bigger issue here is still that we have a recursion issue that i don't understand. and this happens regardless of using automodule, autosummary etc.